noobaa-operator v2.0.2 crash in OCS testing

Hey,

I caught this in the ocs-operator install earlier today. We're using the `image: noobaa/noobaa-operator:2.0.2` version of the operator.

Maybe this is already a known and fixed issue? Below are some logs. 

```
[dvossel@localhost ocs-operator]$ oc get pods -n openshift-storage
NAME                                                              READY   STATUS             RESTARTS   AGE
csi-cephfsplugin-588zf                                            3/3     Running            0          11m
csi-cephfsplugin-7npzd                                            3/3     Running            0          11m
csi-cephfsplugin-pdqnv                                            3/3     Running            0          11m
csi-cephfsplugin-provisioner-57f65684f4-55kw8                     4/4     Running            0          11m
csi-cephfsplugin-provisioner-57f65684f4-tmtrd                     4/4     Running            0          11m
csi-rbdplugin-mcg9c                                               3/3     Running            0          11m
csi-rbdplugin-ntwhf                                               3/3     Running            0          11m
csi-rbdplugin-pkhxx                                               3/3     Running            0          11m
csi-rbdplugin-provisioner-54985c744b-mfssf                        5/5     Running            0          11m
csi-rbdplugin-provisioner-54985c744b-zs766                        5/5     Running            0          11m
local-storage-operator-bcfd5765f-q8kjk                            1/1     Running            0          12m
noobaa-core-0                                                     2/2     Running            0          11m
noobaa-operator-7c55776bf9-4zpd5                                  0/1     CrashLoopBackOff   5          11m
ocs-operator-967957d84-27vx5                                      1/1     Running            0          11m
rook-ceph-drain-canary-ip-10-0-141-176-7485cb45c9-vh5bc           1/1     Running            0          7m17s
rook-ceph-drain-canary-ip-10-0-141-210-dcdd7bf86-ht8fk            1/1     Running            0          7m11s
rook-ceph-drain-canary-ip-10-0-146-186-6b4765894f-hr6l8           1/1     Running            0          6m57s
rook-ceph-mds-test-storagecluster-cephfilesystem-a-8555567sq2fq   1/1     Running            0          6m57s
rook-ceph-mds-test-storagecluster-cephfilesystem-b-548845c2rbr5   1/1     Running            0          6m56s
rook-ceph-mgr-a-d78fd55cc-djnwd                                   1/1     Running            0          8m38s
rook-ceph-mon-a-84575556bb-nmmrf                                  1/1     Running            0          10m
rook-ceph-mon-b-5bd64969ff-grrhc                                  1/1     Running            0          9m44s
rook-ceph-mon-c-55f496cd9d-b4g6c                                  1/1     Running            0          9m10s
rook-ceph-operator-8444cfdc4c-z2k6t                               1/1     Running            0          11m
rook-ceph-osd-0-7488d6cdbd-q2rsd                                  1/1     Running            0          7m22s
rook-ceph-osd-1-6468f8564b-zg7vw                                  1/1     Running            0          7m22s
rook-ceph-osd-2-7d74596989-8hw5k                                  1/1     Running            0          7m18s
rook-ceph-osd-prepare-example-deviceset-0-0-tswwf-fvhql           0/1     Completed          0          8m21s
rook-ceph-osd-prepare-example-deviceset-1-0-v8wbc-pn6sl           0/1     Completed          0          8m21s
rook-ceph-osd-prepare-example-deviceset-2-0-drkcc-kt4jc           0/1     Completed          0          8m20s
rook-ceph-rgw-test-storagecluster-cephobjectstore-a-5758b9m6x8l   1/1     Running            0          6m37s
[dvossel@localhost ocs-operator]$ oc log -f -n openshift-storage noobaa-operator-7c55776bf9-4zpd5
log is DEPRECATED and will be removed in a future version. Use logs instead.
time="2019-10-16T14:35:25Z" level=info msg="CLI version: 2.0.2\n"
time="2019-10-16T14:35:25Z" level=info msg="noobaa-image: noobaa/noobaa-core:5\n"
time="2019-10-16T14:35:25Z" level=info msg="operator-image: noobaa/noobaa-operator:2.0.2\n"
time="2019-10-16T14:35:25Z" level=info msg="Registering Components."
time="2019-10-16T14:35:25Z" level=info msg="OBC Provisioner - start.." provisioner=openshift-storage.noobaa.io/obc
time="2019-10-16T14:35:25Z" level=info msg="running noobaa provisioner openshift-storage.noobaa.io/obc" provisioner=openshift-storage.noobaa.io/obc
time="2019-10-16T14:35:25Z" level=info msg="Starting the Operator ..."
I1016 14:35:25.458502       1 manager.go:118] objectbucket.io/provisioner-manager "msg"="starting provisioner"  "name"="openshift-storage.noobaa.io/obc"
time="2019-10-16T14:35:25Z" level=info msg="Start ..." sys=openshift-storage/noobaa
time="2019-10-16T14:35:25Z" level=info msg="✅ Exists: NooBaa \"noobaa\"\n"
time="2019-10-16T14:35:25Z" level=info msg="SetPhase: \"Verifying\"" sys=openshift-storage/noobaa
time="2019-10-16T14:35:25Z" level=info msg="Parsed image (NamedTagged) noobaa/noobaa-core:5" func=CheckSystemCR sys=openshift-storage/noobaa
time="2019-10-16T14:35:25Z" level=info msg="Parsed version \"5.0.0\" from image tag \"5\"" func=CheckSystemCR sys=openshift-storage/noobaa
time="2019-10-16T14:35:25Z" level=info msg="SetPhase: \"Creating\"" sys=openshift-storage/noobaa
time="2019-10-16T14:35:25Z" level=info msg="found existing credential request for bucket noobaa-backing-store-4691d878-d8b8-49b1-923a-c597b2329ebc" sys=openshift-storage/noobaa
time="2019-10-16T14:35:26Z" level=info msg="ReconcileObject: Done - unchanged /api/v1/namespaces/openshift-storage/secrets/noobaa-server" sys=openshift-storage/noobaa
time="2019-10-16T14:35:26Z" level=info msg="ReconcileObject: Done - updated /apis/apps/v1/namespaces/openshift-storage/statefulsets/noobaa-core" sys=openshift-storage/noobaa
time="2019-10-16T14:35:26Z" level=info msg="ReconcileObject: Done - unchanged /api/v1/namespaces/openshift-storage/services/noobaa-mgmt" sys=openshift-storage/noobaa
time="2019-10-16T14:35:26Z" level=info msg="ReconcileObject: Done - unchanged /api/v1/namespaces/openshift-storage/services/s3" sys=openshift-storage/noobaa
time="2019-10-16T14:35:26Z" level=info msg="SetPhase: \"Connecting\"" sys=openshift-storage/noobaa
time="2019-10-16T14:35:26Z" level=info msg="Collected addresses: &{NodePorts:[https://10.0.141.210:30922] PodPorts:[https://10.128.2.22:8443] InternalIP:[https://172.30.183.27:443] InternalDNS:[https://noobaa-mgmt.openshift-storage:443] ExternalIP:[] ExternalDNS:[https://adf7be5adf02011e99dfc0eca840764d-1672877859.us-east-1.elb.amazonaws.com:443]}" func=CheckServiceStatus service=noobaa-mgmt sys=openshift-storage/noobaa
time="2019-10-16T14:35:26Z" level=info msg="Collected addresses: &{NodePorts:[https://10.0.141.210:30176] PodPorts:[https://10.128.2.22:6443] InternalIP:[https://172.30.180.124:443] InternalDNS:[https://s3.openshift-storage:443] ExternalIP:[] ExternalDNS:[https://adf814de9f02011e99dfc0eca840764d-330692352.us-east-1.elb.amazonaws.com:443]}" func=CheckServiceStatus service=s3 sys=openshift-storage/noobaa
time="2019-10-16T14:35:26Z" level=info msg="✈️  RPC: auth.read_auth() Request: <nil>"
time="2019-10-16T14:35:26Z" level=info msg="✅ RPC: auth.read_auth() Response OK: &{Op:res RequestID:1@https://::ffff:10.128.2.1:42620(2j3c4uro) Took:0.11099199997261167 Error:<nil>}"
time="2019-10-16T14:35:26Z" level=info msg="SetPhase: \"Configuring\"" sys=openshift-storage/noobaa
time="2019-10-16T14:35:26Z" level=info msg="✅ Exists: Secret \"noobaa-operator\"\n"
time="2019-10-16T14:35:26Z" level=info msg="✅ Exists: Secret \"noobaa-admin\"\n"
time="2019-10-16T14:35:26Z" level=info msg="❌ Not Found: BackingStore \"noobaa-default-backing-store\"\n"
time="2019-10-16T14:35:26Z" level=info msg="✅ Exists:  \"noobaa-cloud-creds-secret\"\n"
time="2019-10-16T14:35:26Z" level=info msg="Secret noobaa-cloud-creds-secret was created succesfully by cloud-credentials operator" func=ReconcileDefaultBackingStore sys=openshift-storage/noobaa
time="2019-10-16T14:35:26Z" level=panic msg="☠️  Panic Attack: [] failed to parse AWS region from node name ip-10-0-141-139.ec2.internal"
E1016 14:35:26.117429       1 runtime.go:69] Observed a panic: &logrus.Entry{Logger:(*logrus.Logger)(0xc0000ee150), Data:logrus.Fields{}, Time:time.Time{wall:0xbf61e88b86fdb5b1, ext:1159227328, loc:(*time.Location)(0x289f2c0)}, Level:0x0, Caller:(*runtime.Frame)(nil), Message:"☠️  Panic Attack: [] failed to parse AWS region from node name ip-10-0-141-139.ec2.internal", Buffer:(*bytes.Buffer)(nil), Context:(*context.emptyCtx)(0xc0000c0210), err:""} (&{0xc0000ee150 map[] 2019-10-16 14:35:26.117290417 +0000 UTC m=+1.159227328 panic <nil> ☠️  Panic Attack: [] failed to parse AWS region from node name ip-10-0-141-139.ec2.internal <nil> context.TODO })
noobaa-operator/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:76
noobaa-operator/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:65
noobaa-operator/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:51
/usr/local/Cellar/go/1.13.1/libexec/src/runtime/panic.go:679
noobaa-operator/vendor/github.com/sirupsen/logrus/entry.go:239
noobaa-operator/vendor/github.com/sirupsen/logrus/entry.go:268
noobaa-operator/vendor/github.com/sirupsen/logrus/entry.go:314
noobaa-operator/vendor/github.com/sirupsen/logrus/entry.go:352
noobaa-operator/pkg/util/util.go:401
noobaa-operator/pkg/util/util.go:669
noobaa-operator/pkg/system/phase4_configuring.go:208
noobaa-operator/pkg/system/phase4_configuring.go:39
noobaa-operator/pkg/system/reconciler.go:226
noobaa-operator/pkg/system/reconciler.go:187
noobaa-operator/pkg/controller/noobaa/noobaa_controller.go:31
noobaa-operator/vendor/sigs.k8s.io/controller-runtime/pkg/reconcile/reconcile.go:92
noobaa-operator/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:215
noobaa-operator/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:158
noobaa-operator/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:133
noobaa-operator/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:134
noobaa-operator/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:88
/usr/local/Cellar/go/1.13.1/libexec/src/runtime/asm_amd64.s:1357
panic: (*logrus.Entry) (0x190ad40,0xc0001863f0) [recovered]
	panic: (*logrus.Entry) (0x190ad40,0xc0001863f0)

goroutine 361 [running]:
k8s.io/apimachinery/pkg/util/runtime.HandleCrash(0x0, 0x0, 0x0)
	noobaa-operator/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:58 +0x105
panic(0x190ad40, 0xc0001863f0)
	/usr/local/Cellar/go/1.13.1/libexec/src/runtime/panic.go:679 +0x1b2
github.com/sirupsen/logrus.Entry.log(0xc0000ee150, 0xc0003c2ff0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
	noobaa-operator/vendor/github.com/sirupsen/logrus/entry.go:239 +0x2db
github.com/sirupsen/logrus.(*Entry).Log(0xc000168850, 0x0, 0xc00142d670, 0x1, 0x1)
	noobaa-operator/vendor/github.com/sirupsen/logrus/entry.go:268 +0xeb
github.com/sirupsen/logrus.(*Entry).Logf(0xc000168850, 0x0, 0x1958c31, 0x1d, 0xc00142d6d0, 0x2, 0x2)
	noobaa-operator/vendor/github.com/sirupsen/logrus/entry.go:314 +0xe2
github.com/sirupsen/logrus.(*Entry).Panicf(...)
	noobaa-operator/vendor/github.com/sirupsen/logrus/entry.go:352
github.com/noobaa/noobaa-operator/v2/pkg/util.Panic(0x1c09ca0, 0xc000d00540)
	noobaa-operator/pkg/util/util.go:401 +0xf3
github.com/noobaa/noobaa-operator/v2/pkg/util.GetAWSRegion(0x166bfc0, 0xc000d37030)
	noobaa-operator/pkg/util/util.go:669 +0x621
github.com/noobaa/noobaa-operator/v2/pkg/system.(*Reconciler).ReconcileDefaultBackingStore(0xc000b62120, 0x0, 0x0)
	noobaa-operator/pkg/system/phase4_configuring.go:208 +0x2d5
github.com/noobaa/noobaa-operator/v2/pkg/system.(*Reconciler).ReconcilePhaseConfiguring(0xc000b62120, 0x0, 0x0)
	noobaa-operator/pkg/system/phase4_configuring.go:39 +0xbe
github.com/noobaa/noobaa-operator/v2/pkg/system.(*Reconciler).ReconcilePhases(0xc000b62120, 0xc00068c700, 0x193a101)
	noobaa-operator/pkg/system/reconciler.go:226 +0x8a
github.com/noobaa/noobaa-operator/v2/pkg/system.(*Reconciler).Reconcile(0xc000b62120, 0x11, 0xc000741fb4, 0x6, 0x1c5b280)
	noobaa-operator/pkg/system/reconciler.go:187 +0xbf
github.com/noobaa/noobaa-operator/v2/pkg/controller/noobaa.Add.func1(0xc0005522e0, 0x11, 0xc000741fb4, 0x6, 0xc0001c5270, 0x42d0aa, 0xc00073c7e0, 0x36baef608900)
	noobaa-operator/pkg/controller/noobaa/noobaa_controller.go:31 +0x110
sigs.k8s.io/controller-runtime/pkg/reconcile.Func.Reconcile(0xc00077aca0, 0xc0005522e0, 0x11, 0xc000741fb4, 0x6, 0x0, 0x0, 0x0, 0x0)
	noobaa-operator/vendor/sigs.k8s.io/controller-runtime/pkg/reconcile/reconcile.go:92 +0x4e
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem(0xc0001ae0a0, 0x0)
	noobaa-operator/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:215 +0x20a
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func1()
	noobaa-operator/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:158 +0x36
k8s.io/apimachinery/pkg/util/wait.JitterUntil.func1(0xc000780030)
	noobaa-operator/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:133 +0x5e
k8s.io/apimachinery/pkg/util/wait.JitterUntil(0xc000780030, 0x3b9aca00, 0x0, 0x1, 0xc000369620)
	noobaa-operator/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:134 +0xf8
k8s.io/apimachinery/pkg/util/wait.Until(0xc000780030, 0x3b9aca00, 0xc000369620)
	noobaa-operator/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:88 +0x4d
created by sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start
	noobaa-operator/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:157 +0x32e
```

--- a/pkg/system/phase4_configuring.go
+++ b/pkg/system/phase4_configuring.go
@@ -205,7 +205,12 @@ func (r *Reconciler) ReconcileDefaultBackingStore() error {
 	log.Infof("Secret %s was created succesfully by cloud-credentials operator", r.CloudCreds.Spec.SecretRef.Name)
 
 	// create the acutual S3 bucket
-	region := util.GetAWSRegion()
+	region, err := util.GetAWSRegion()
+	if err != nil {
+		r.Recorder.Eventf(r.NooBaa, corev1.EventTypeWarning, "DefaultBackingStoreFailure",
+			"Failed to get AWSRegion. using	 us-east-1 as the default region. %q", err)
+		region = "us-east-1"
+	}
 	r.Logger.Infof("identified aws region %s", region)
 	s3Config := &aws.Config{
 		Credentials: credentials.NewStaticCredentials(

--- a/pkg/util/util.go
+++ b/pkg/util/util.go
@@ -638,37 +638,45 @@ func IsAWSPlatform() bool {
 }
 
 // GetAWSRegion parses the region from a node's name
-func GetAWSRegion() string {
-	var validAWSRegions = map[string]bool{
-		"us-east-1":      true,
-		"us-east-2":      true,
-		"us-west-1":      true,
-		"us-west-2":      true,
-		"ca-central-1":   true,
-		"eu-central-1":   true,
-		"eu-west-1":      true,
-		"eu-west-2":      true,
-		"eu-west-3":      true,
-		"eu-north-1":     true,
-		"ap-east-1":      true,
-		"ap-northeast-1": true,
-		"ap-northeast-2": true,
-		"ap-northeast-3": true,
-		"ap-southeast-1": true,
-		"ap-southeast-2": true,
-		"ap-south-1":     true,
-		"me-south-1":     true,
-		"sa-east-1":      true,
+func GetAWSRegion() (string, error) {
+	// parse the node name to get AWS region according to this: 
+	// https://docs.aws.amazon.com/en_pv/vpc/latest/userguide/vpc-dns.html#vpc-dns-hostnames
+	var mapValidAWSRegions = map[string]string{
+		"compute-1":      "us-east-1",
+		"ec2":            "us-east-1",
+		"us-east-1":      "us-east-1",
+		"us-east-2":      "us-east-2",
+		"us-west-1":      "us-west-1",
+		"us-west-2":      "us-west-2",
+		"ca-central-1":   "ca-central-1",
+		"eu-central-1":   "eu-central-1",
+		"eu-west-1":      "eu-west-1",
+		"eu-west-2":      "eu-west-2",
+		"eu-west-3":      "eu-west-3",
+		"eu-north-1":     "eu-north-1",
+		"ap-east-1":      "ap-east-1",
+		"ap-northeast-1": "ap-northeast-1",
+		"ap-northeast-2": "ap-northeast-2",
+		"ap-northeast-3": "ap-northeast-3",
+		"ap-southeast-1": "ap-southeast-1",
+		"ap-southeast-2": "ap-southeast-2",
+		"ap-south-1":     "ap-south-1",
+		"me-south-1":     "me-south-1",
+		"sa-east-1":      "sa-east-1",
 	}
 	nodesList := &corev1.NodeList{}
 	if ok := KubeList(nodesList, nil); !ok || len(nodesList.Items) == 0 {
-		Panic(fmt.Errorf("failed to list kubernetes nodes"))
+		return "", fmt.Errorf("Failed to list kubernetes nodes")
 	}
 	nameSplit := strings.Split(nodesList.Items[0].Name, ".")
-	if len(nameSplit) < 2 || !validAWSRegions[nameSplit[1]] {
-		Panic(fmt.Errorf("failed to parse AWS region from node name %s", nodesList.Items[0].Name))
+	if len(nameSplit) < 2 {
+		return "", fmt.Errorf("Unexpected node name format: %q", nodesList.Items[0].Name)
 	}
-	return nameSplit[1]
+	awsRegion := mapValidAWSRegions[nameSplit[1]]
+	if awsRegion == "" {
+		return "", fmt.Errorf("The parsed AWS region is invalid: %q", awsRegion)
+	}
+	return awsRegion, nil
 }
 
 // GetFlagStringOrPrompt returns flag value but if empty will promtp to read from stdin

